frontend/packages/enduser-frontend/src/views/applications/daycare-application/daycare-application.vue

```diff
@@ -1076,509 +1087,523 @@ SPDX-License-Identifier: LGPL-2.1-or-later
-          (validation) => validation.errors.length > 0
-        )
-
-        let formIsValid = this.formIsValid
-
-        _.forEach(errors, (error) => {
-          const key = error.errors[0].key
-          const simpleMessage = error.errors[0].simpleMessage
-          const validationName = error.errors[0].validationName
-          const findKey = _.findKey(this.validations, key) as string
-          if (findKey) {
-            this.errors[findKey].push(
-              validationName + this.$t(simpleMessage).toString()
-            )
-            formIsValid = false
-          }
-        })
-        if (this.preferredUnits.length === 0) {
-          this.errors.preferredUnits.push(this.preferredUnitsValidationText())
+    onPreparatoryChanged() {
+      this.reloadUnits()
+    },
+    preferredUnitsValidationText() {
+      return this.$t(
+        'form.daycare-application.preferredUnits.unit-required-validation'
+      ).toString()
+    },
+    serviceTimeValidationText() {
+      return (
+        this.$t(
+          'form.daycare-application.service.dailyTime.label'
+        ).toString() +
+        this.$t('validation.errors.is-required-field').toString()
+      )
+    },
+    onSiblingBasisClicked() {
+      if (!this.model.apply.siblingBasis) {
+        this.model.apply.siblingName = ''
+        this.model.apply.siblingSsn = ''
+      }
+    },
+    validateForm() {
+      this.debouncedValidate.cancel()
+      this.errors = createDaycareSectionError()
+      const form = this.$refs.form as any
+      const validate = form.validate()
+      const errors = _.filter(
+        validate,
+        (validation) => validation.errors.length > 0
+      )
+
+      let formIsValid = this.formIsValid
+
+      _.forEach(errors, (error) => {
+        const key = error.errors[0].key
+        const simpleMessage = error.errors[0].simpleMessage
+        const validationName = error.errors[0].validationName
+        const findKey = _.findKey(this.validations, key) as string
+        if (findKey) {
+          this.errors[findKey].push(
+            validationName + this.$t(simpleMessage).toString()
+          )
           formIsValid = false
         }
+      })
+      if (this.preferredUnits.length === 0) {
+        this.errors.preferredUnits.push(this.preferredUnitsValidationText())
+        formIsValid = false
+      }
 
-        if (
-          this.showDailyTime &&
-          (!this.model.serviceStart || !this.model.serviceEnd)
-        ) {
-          this.errors.service.push(this.serviceTimeValidationText())
-          formIsValid = false
-        }
+      if (
+        this.showDailyTime &&
+        (!this.model.serviceStart || !this.model.serviceEnd)
+      ) {
+        this.errors.service.push(this.serviceTimeValidationText())
+        formIsValid = false
+      }
 
+      if (this.hasOutsidePreschoolTermError) {
+        this.errors.service.push(this.$t('validation.errors.outside-preschool-term').toString())
+        formIsValid = false
```

Comment:
I was having a pretty decent day until I learnt this is how we handle validations here :man_facepalming: (this is OK for this change, just gotta do better in the rewrite).

Reply:
Yeah I spend couple hours trying to understand this mess without success.. In the end I just copied the simplest solution. Hopefully this will all be gone soon.